### PR TITLE
Implementation for fallbacks for empty email-address fields.

### PR DESCRIPTION
**Pull request by Zerosan**
Prevents phpmail error due to empty adminmail and falls back to sendmail_from value if adminmail and return mail are missing.

### CODE REVIEW

**Comment by JN-Jones** [comment]
@mybb/developers @mybb/sqa What shall we do with this? IIRC someone mentioned on slack that this isn't necessary?

**Comment by Zerosan** [comment]
As it is the current system overwrites an existing sendmail_from with an empty value, which seems quite suboptimal. 
An alternative would be to make the admin mail required,which wouldn't have any impact on existing installations with this configuration error. 
Obviously in a correctly and completely setup Mybb installation this is not necessary.

**Comment by JN-Jones** [comment]
The installer makes sure that the adminemail is set AFAIK (haven't checked but from a quick look at the code it should throw an error)

**Comment by Shade-** [comment]
If my interpretation is correct, the installer checks if the admin's email is empty and throws an error if it is; the adminemail setting inherits the value upon installation, but can be changed at a later stage from the ACP, which does not check its emptiness, leading to potential blank values and errors by php's mail() function.

**Comment by Ben-Conway** [comment]
@mybb/developers @mybb/sqa What should we do with this PR?

**Comment by Zerosan** [comment]
To me it appears akin to a bug to interfere with the existing php mail settings when mybb doesn't have a valid value to supply itself.

This can be a case of "leave empty to use php mail default"

**Comment by WildcardSearch** [comment]
This seems like a legitimate issue. Would it be a waste of my time to review the changes? There code needs an edit or two before it can be merged.

**Comment by Ben-Conway** [comment]
If it's a legitimate issue than do we want it included in 1.8.10 @euantorano ?

**Comment by Stefan-ST** [comment]
> This can be a case of "leave empty to use php mail default"
On the other hand the PR doesn't provide this. While MyBB will still send emails it needs the admin email for other features.

**Comment by euantorano** [comment]
This isn't feature complete, nor does it conform to our code standards with regards to the placement of curly brackets.

**Comment by euantorano** [comment]
In addition to @Stefan-ST's comment, the code doesn't conform to our code style standards.

**Comment by effone** [comment]
As I don't think @Zerosan is going to do anything in this PR we should close this, create an issue for this, write proper PR and merge. @mybb/developers ?

**Comment by euantorano** [comment]
Sounds like the best way ahead to me, but we should probably leave this until 1.8.17 I think as it might need quite a bit of testing.

File: inc/class_mailhandler.php [hunk 1]
```diff
@@ -118,9 +118,11 @@ function get_from_email()
 		{
 			$email = $mybb->settings['returnemail'];
 		}
-		else
+		else if(trim($mybb->settings['adminemail']))
 		{
 			$email = $mybb->settings['adminemail'];
+		}else{
```
[the remark this answers is not in the document]
Comment (cloudenv) [reply inc/class_mailhandler.php]: `} else {`

**Comment by Sama34** [comment]
> I can confirm `$mybb->settings['adminemail']` can be empty.

Then whether the installer should stop forcing it or the ACP should start doing it.

**Comment by euantorano** [comment]
IMO the ACP should start forcing it to be a valid email address.

**Comment by Ben-MyBB** [comment]
https://github.com/mybb/mybb/issues/3955

**Pull request closed without merging**